lib/jsdom/level2/html.js

```diff
@@ -2048,68 +2048,72 @@ function loadFrame (frame) {
+
+    if (frame._parentNode) {
+      defineGetter(window, name, function () { return frame.contentWindow; });
+    }
+  }
+}
+
 define('HTMLFrameElement', {
   tagName: 'FRAME',
-  init : function () {
-    // Set up the frames array.  window.frames really just returns a reference
-    // to the window object, so the frames array is just implemented as indexes
-    // on the window.
-    var parent = this._ownerDocument.defaultView;
-    var frameID = parent._length++;
-    var self = this;
-    defineGetter(parent, frameID, function () {
-      return self.contentWindow;
-    });
-
-    // The contentDocument/contentWindow shouldn't be created until the frame
-    // is inserted:
-    // "When an iframe element is first inserted into a document, the user
-    //  agent must create a nested browsing context, and then process the
-    //  iframe attributes for the first time."
-    //  (http://www.whatwg.org/specs/web-apps/current-work/#the-iframe-element)
-    this._initInsertListener = function () {
-      loadFrame(self);
-    };
-    this.addEventListener('DOMNodeInsertedIntoDocument', this._initInsertListener, false);
-  },
   proto: {
-    _attrModified: function(name, value, oldVal) {
+    _attrModified: function (name, value, oldVal) {
       core.HTMLElement.prototype._attrModified.call(this, name, value, oldVal);
-      var self = this;
       if (name === 'name') {
-        // Remove named frame access.
         if (oldVal) {
-          this._ownerDocument.defaultView._frame(oldVal);
-        }
-        // Set up named frame access.
-        if (value) {
-          this._ownerDocument.defaultView._frame(value, this);
+          // I do not know why this only works with _global and not with _defaultView :(
+          delete this._ownerDocument._global[oldVal];
         }
+        refreshNameAccessor(this);
       } else if (name === 'src') {
-        // Page we don't fetch the page until the node is inserted. This at
-        // least seems to be the way Chrome does it.
-        if (!this._attachedToDocument) {
-          if (!this._waitingOnInsert) {
-            // First, remove the listener added in 'init'.
-            this.removeEventListener('DOMNodeInsertedIntoDocument',
-                                     this._initInsertListener, false)
-
-            // If we aren't already waiting on an insert, add a listener.
-            // This guards against src being set multiple times before the frame
-            // is inserted into the document - we don't want to register multiple
-            // callbacks.
-            this.addEventListener('DOMNodeInsertedIntoDocument', function loader () {
-              self.removeEventListener('DOMNodeInsertedIntoDocument', loader, false);
-              this._waitingOnInsert = false;
-              loadFrame(self);
-            }, false);
-            this._waitingOnInsert = true;
-          }
-        } else {
-          loadFrame(self);
+        if (this._parentNode) {
```

Comment:
I think the intent of all the crazy above this was to never load the frame until it was appended to body, so, should be equivalent? But yeah, worth checking to see if existing tests cover it and if not adding more and making sure the results are the same; will do.